keystone build fails because -lc++ flag used

In this commit https://github.com/radare/radare2-extras/commit/467d4785ce8f341371cffb52b59127a8bbac478f flag ``-lc++`` is added, and compiling fails:
````
$ r2pm -i keystone
r_config_set: variable 'file.id' not found
r_config_set: variable 'file.flag' not found
r_config_set: variable 'file.analyze' not found
Processing keystone ...
Already up-to-date.
Install Done For keystone
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for working directories... current
using prefix '/home/user/.config/radare2/prefix/'
checking for c compiler... gcc
checking for c++ compiler... g++
checking for libkeystone ... yes
Using PKGCONFIG: pkg-config
checking pkg-config flags for r_asm... yes
creating ./config.mk
cleaning temporally files... done
rm -f *.so
make asm_arm_ks.so
make[1]: Entering directory `/home/user/.config/radare2/r2pm/git/radare2-extras/keystone'
gcc -lc++ -fPIC -shared -Wall -I/usr/include/libr -lr_asm -lr_syscall -lr_lang -lr_util -lr_parse -lr_flags -lr_cons -lr_reg -lr_anal -I/usr/local/include   -L/usr/local/lib -lkeystone   -o asm_arm_ks.so asm_arm_ks.c -lkeystone
/usr/bin/ld: cannot find -lc++
collect2: error: ld returned 1 exit status
make[1]: *** [asm_arm_ks.so] Error 1
make[1]: Leaving directory `/home/user/.config/radare2/r2pm/git/radare2-extras/keystone'
make: *** [arm] Error 2
````
I had to edit Makefile and remove this ``-lc++`` flag. Maybe on some OS-es this flag is important
OS: Ubuntu 14.04.5 LTS if this matters

Did u r2pm update?

> On 22 Oct 2016, at 18:24, szt <notifications@github.com> wrote:
> 
> In this commit 467d478
> 
> $ r2pm -i keystone
> r_config_set: variable 'file.id' not found
> r_config_set: variable 'file.flag' not found
> r_config_set: variable 'file.analyze' not found


Clean your radare2rc

> Processing keystone ...
> Already up-to-date.
> Install Done For keystone
> checking build system type... x86_64-unknown-linux-gnu
> checking host system type... x86_64-unknown-linux-gnu
> checking target system type... x86_64-unknown-linux-gnu
> checking for working directories... current
> using prefix '/home/user/.config/radare2/prefix/'
> checking for c compiler... gcc
> checking for c++ compiler... g++
> checking for libkeystone ... yes
> Using PKGCONFIG: pkg-config
> checking pkg-config flags for r_asm... yes
> creating ./config.mk
> cleaning temporally files... done
> rm -f *.so
> make asm_arm_ks.so
> make[1]: Entering directory `/home/user/.config/radare2/r2pm/git/radare2-extras/keystone'
> gcc -lc++ -fPIC -shared -Wall -I/usr/include/libr -lr_asm -lr_syscall -lr_lang -lr_util -lr_parse -lr_flags -lr_cons -lr_reg -lr_anal -I/usr/local/include   -L/usr/local/lib -lkeystone   -o asm_arm_ks.so asm_arm_ks.c -lkeystone
> /usr/bin/ld: cannot find -lc++
> collect2: error: ld returned 1 exit status
> make[1]: *** [asm_arm_ks.so] Error 1
> make[1]: Leaving directory `/home/user/.config/radare2/r2pm/git/radare2-extras/keystone'
> make: *** [arm] Error 2
> I had to edit Makefile and remove this -lc++ flag. Maybe on sime OS-es this flag is important
> OS: Ubuntu 14.04.5 LTS if this matter
> 

Can you send a pr?
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

>Did u r2pm update?
...
>Clean your radare2rc

This may not affect this issue. If you try to do compile some c++ code with ``-lc++`` on some linux distro without libcxx, it will fail.
````
$ cat text.c++ 
#include <iostream>
int main(){ std::cout << 1; return 0;}
$ g++ -lc++ text.c++
/usr/bin/ld: cannot find -lc++
collect2: error: ld returned 1 exit status
````

so what do you propose? tell the user in a human way that their system is fuckedup?

i think debian is the only stupid distro that allows to install g++ without libcxx-dev as a dependency

> On 26 Oct 2016, at 19:47, szt <notifications@github.com> wrote:
> 
> Did u r2pm update?
> ...
> Clean your radare2rc
> 
> This may not affect this issue. If you try to do compile some c++ code with -lc++ on some linux distro without libcxx, it will fail.
> 
> $ cat text.c++ 
> #include <iostream>
> int main(){ std::cout << 1; return 0;}
> $ g++ -lc++ text.c++
> /usr/bin/ld: cannot find -lc++
> collect2: error: ld returned 1 exit status
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub <https://github.com/radare/radare2-extras/issues/70#issuecomment-256424977>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AA3-lkJq5RDVu6b1Ii8Ic7kxJUGPqXk-ks5q35IggaJpZM4Kd5tA>.
>

--- a/keystone/Makefile
+++ b/keystone/Makefile
@@ -13,10 +13,13 @@ all: $(PLUGINS)
 #$(info $(LIBS))
 
 define kscc
-$(CC) -lc++ -fPIC $(LDFLAGS_SHARED) -Wall\
-	$(R2_CFLAGS) $(R2_LDFLAGS)\
-	$(KS_CFLAGS) $(KS_LDFLAGS)\
-	-o asm_$(1)_ks.$(SO_EXT) asm_$(1)_ks.c $(KS_LINK)
+$(CC) -c -fPIC $(LDFLAGS_SHARED) -Wall\
+	$(R2_CFLAGS) $(KS_CFLAGS)\
+	-o asm_$(1)_ks.$(O_EXT) asm_$(1)_ks.c
+$(CXX)  $(LDFLAGS_SHARED) -Wall\
+	$(R2_LDFLAGS) $(KS_LDFLAGS)\
+	-o asm_$(1)_ks.$(SO_EXT) asm_$(1)_ks.$(O_EXT) $(KS_LINK)
+rm asm_$(1)_ks.$(O_EXT)
 endef
 
 define ksmake

--- a/keystone/Makefile
+++ b/keystone/Makefile
@@ -13,10 +13,10 @@ all: $(PLUGINS)
 #$(info $(LIBS))
 
 define kscc
-$(CC) -c -fPIC $(LDFLAGS_SHARED) -Wall\
+$(CC) -c -fPIC -Wall\
 	$(R2_CFLAGS) $(KS_CFLAGS)\
 	-o asm_$(1)_ks.$(O_EXT) asm_$(1)_ks.c
-$(CXX)  $(LDFLAGS_SHARED) -Wall\
+$(CXX) $(LDFLAGS_SHARED)\
 	$(R2_LDFLAGS) $(KS_LDFLAGS)\
 	-o asm_$(1)_ks.$(SO_EXT) asm_$(1)_ks.$(O_EXT) $(KS_LINK)
 rm asm_$(1)_ks.$(O_EXT)

--- a/keystone/build.mk
+++ b/keystone/build.mk
@@ -11,6 +11,7 @@ else
 SO_EXT=so
 endif
 endif
+O_EXT=o
 
 KS_CFLAGS=$(shell pkg-config --cflags keystone)
 KS_LDFLAGS=$(shell pkg-config --libs keystone)

